AppCenter/AppCenter/Internals/Util/MSUtility+Application.m

```diff
@@ -47,6 +47,11 @@ + (UIApplication *)sharedApp {
 
 #if TARGET_OS_OSX
 + (MSApplicationState)sharedAppState {
+
+  // UI API (isHidden) cannot be called from a background thread.
+  if (![NSThread isMainThread]) {
+    return MSApplicationStateUnknown;
```

Comment:
The problem is now the caller has to make sure to be running on the main thread to have a reliable result. Shouldn't we make it async and dispatch to main thread on macOS? It's more change but less error prone in the usage of that API.

Reply:
> Shouldn't we make it async and dispatch to main thread on macOS?

No. The main thread is blocked at that moment.

It's fine to return `MSApplicationStateUnknown` at this case, since it's will happen only in one place where we can proceed without this check. I've left the comment there.